docs/docsite/rst/dev_guide/developing_modules_general_windows.rst

```diff
@@ -0,0 +1,256 @@
+    }
+
+    Import-Module -Name .\Ansible.ModuleUtils.PowershellLegacy.psm1
+    ### end setup code
+
+You can add more args to ``$complex_args`` as required by the module. The
+module can now be run on the remote server either directly through Powershell
+or through an IDE.
+
+There are multiple IDE's that can be used to debug a Powershell script, two of
+the most popular are
+
+- `Powershell ISE`_
+- `Visual Studio Code`_
+
+.. _Powershell ISE: https://msdn.microsoft.com/en-us/powershell/scripting/core-powershell/ise/how-to-debug-scripts-in-windows-powershell-ise
+.. _Visual Studio Code: https://blogs.technet.microsoft.com/heyscriptingguy/2017/02/06/debugging-powershell-script-in-visual-studio-code-part-1/
+
+To be able to view the arguments as passed by Ansible to the module follow
+these steps.
+
+- Before executing the Ansible command run ``$ export ANSIBLE_KEEP_REMOTE_FILES=1`` to get Ansible to keep the exec files on the server
+- Run the Ansible command
+- Run ``$ export ANSIBLE_KEEP_REMOTE_FILES=0`` to stop Ansible from filling up the temp space on the server
+- Log onto the Windows server using the same user Ansible executed the module as
+- Navigate to ``%TEMP%\..``, there should be a folder starting with ``ansible-temp-``
+- Inside this folder open up the powershell script for the module
+- In this script there is a raw JSON script under ``$json_raw`` which contains the module arguments under ``module_args``
+- These args can be assigned manually to the ``$complex_args`` variable that is defined on your debug script
+
+
+Unit testing
+============
+
+Currently there is no mechanism to develop unit tests for Powershell modules.
+There is work in the pipeline to introduce this in the future, stay tuned.
+
+
+Integration testing
+===================
+
+Integration tests for modules will be appropriately located in
+``./test/integration/targets``. You must first set up your testing environment
+and configure a test inventory for Ansible to connect to. In this example we
+will setup a test inventory to connect to two hosts and run the integration
+tests for win_stat.
+
+- Create a copy of ``./test/integration/inventory.winrm.template`` and just call it ``inventory.winrm``
+- Fill in entries under ``[windows]`` and set the required vars that are needed to connect to the host
+- To run the integration tests run ``ansible-test windows-integration win_stat``, you can replace ``win_stat`` with the role you wish to test
+
+This will go through all the tests currently written for that role. You can set
+the verbosity level using the ``-v`` argument just as you would with
+ansible-playbook.
+
+When developing tests for a new module, it is recommended to test a scenario in
+check mode and 2 times not in check mode. This ensure we test that check mode
+does not make any changes but reports a change as well as the 2nd run stays
+idempotent. This is an example of one way that this can be done
+
+::
+
+    # check mode tests for scenario
```

Comment:
Sounds good, I've added (check mode) and (idempotent) to the relevant tests, good suggestions